algorithms/unit_tests/TestStdAlgorithmsTeamEqual.cpp

```diff
@@ -0,0 +1,262 @@
+        break;
+      }
+
+      case 2: {
+        const bool result = KE::equal(member, rowData, rowComp);
+        Kokkos::single(Kokkos::PerTeam(member),
+                       [=]() { m_resultsView(rowIndex) = result; });
+        break;
+      }
+
+      case 3: {
+        const bool result = KE::equal(member, rowData, rowComp, m_binaryPred);
+        Kokkos::single(Kokkos::PerTeam(member),
+                       [=]() { m_resultsView(rowIndex) = result; });
+
+        break;
+      }
+
+      case 4: {
+        const bool result =
+            KE::equal(member, dataBegin, dataEnd, compBegin, compEnd);
+        Kokkos::single(Kokkos::PerTeam(member),
+                       [=]() { m_resultsView(rowIndex) = result; });
+        break;
+      }
+
+      case 5: {
+        const bool result = KE::equal(member, dataBegin, dataEnd, compBegin,
+                                      compEnd, m_binaryPred);
+        Kokkos::single(Kokkos::PerTeam(member),
+                       [=]() { m_resultsView(rowIndex) = result; });
+        break;
+      }
+    }
+  }
+};
+
+template <class LayoutTag, class ValueType>
+void test_A(const bool viewsAreEqual, std::size_t numTeams, std::size_t numCols,
+            int apiId) {
+  /* description:
+     use a rank-2 view randomly filled with values,
+     and run a team-level equal
+   */
+
+  // -----------------------------------------------
+  // prepare data
+  // -----------------------------------------------
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  constexpr ValueType lowerBound = 5;
+  constexpr ValueType upperBound = 523;
+  const auto bounds              = make_bounds(lowerBound, upperBound);
+
+  auto [dataView, dataViewBeforeOp_h] = create_random_view_and_host_clone(
+      LayoutTag{}, numTeams, numCols, bounds, "dataView");
+
+  // create a view to compare it with dataView. If viewsAreEqual == true,
+  // compView is a copy of dataView. If viewsAreEqual == false, compView is
+  // randomly filled
+  auto compView   = create_deep_copyable_compatible_clone(dataView);
+  auto compView_h = create_mirror_view(Kokkos::HostSpace(), compView);
+  if (viewsAreEqual) {
+    Kokkos::deep_copy(compView_h, dataViewBeforeOp_h);
+  } else {
+    using rand_pool =
+        Kokkos::Random_XorShift64_Pool<Kokkos::DefaultHostExecutionSpace>;
+    rand_pool pool(lowerBound * upperBound);
+    Kokkos::fill_random(compView_h, pool, lowerBound, upperBound);
+  }
+
+  Kokkos::deep_copy(compView, compView_h);
+
+  // -----------------------------------------------
+  // launch kokkos kernel
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+
+  // create the view to store results of equal()
+  Kokkos::View<bool*> resultsView("resultsView", numTeams);
```

Comment:
Don't you want to set a memory space?

Reply:
I am using the default one here because the data views are always constructed using the default mem space too . makes sense?